api/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/gravitational/teleport/api
 
-go 1.21
+go 1.22
```

Comment:
Is this intentional?

Reply:
Otel now requires Go 1.22, so this gets updated automatically. Unavoidable if we want to move certain dependencies forward.

Reply:
(Well spotted, btw. I meant to add a comment about it but it escaped me.(